lib/galaxy/authnz/managers.py

```diff
@@ -102,8 +105,14 @@ def _parse_idp_config(self, config_xml):
             'client_id': config_xml.find('client_id').text,
             'client_secret': config_xml.find('client_secret').text,
             'redirect_uri': config_xml.find('redirect_uri').text}
+
         if config_xml.find('prompt') is not None:
             rtv['prompt'] = config_xml.find('prompt').text
+        if config_xml.find('api_url') is not None:
+            rtv['api_url'] = config_xml.find('api_url').text
+        if config_xml.find('url') is not None:
```

Comment:
Could you please elaborate on what is this URL?
- it is not set it in the config sample;
- can we compose `api_url` given the `url`?

Reply:
That's for possible future compatibility. In PSA e.g. GitHub Enterprise could make use of this. It's not used right now, but I don't see that it hurts either.

Adding the flexibility however, does seem like a good idea to me.

Reply:
Is `url` a substring of `api_url`? e.g., `url = http://xyz.com/` and `api_url = http://xyz.com/oauth2/default`?

Reply:
Not necessarily. Okta isn't using url. It depends on a per IdP basis.

Reply:
Could you please elaborate a bit more?

Reply:
I opted to create more generic code for adding more IdPs in the future. Unlike. E.g. Google where the domain and path are consistent, this doesn't have to be the case for commercial IdPs. Therefore allowing these values to be set allows more flexibility for possible future expansion of the IdPs if the arises. So this could also be used for GitHub (Enterprise), which for regular github uses 'github.com', but for the API uses 'api.github.com'. So by trying to automatically determine this, we'd need to write code for this flexibility for every IdP that would qualify. So in order to create flexibility, having both of these as configuration instead would result in the most flexible backend with the least amount of code required for specialisation towards a certain IdP.